### ISSUE TITLE
Autocomplete does not work at a depth of 3 groups

### ISSUE THREAD

**Issue by objque** [pallets/click#600]
I have this command structure.

```
cli        (group - entry point)
-ip        (group)
--static   (group)
---foo     (method)
--floating (group)
---bar     (method)
```

And when I press <kbd>`Tab`</kbd><kbd>`Tab`</kbd> keys:
1. attempt - i have correct autocomplete such as `ip` command
2. attempt - i have correct autocomplete such as `floating` and `static` commands
3. attempt - i have **incorrect** autocomplete - list of all files in current dir instead of `bar` command.

[![screenshot](http://i.imgur.com/SisYVgQ.png)](http://i.imgur.com/SisYVgQ.png)

Here's the code:
**shell.py**

```
# coding=utf-8
import click


@click.group()
def cli():
    """Entry point"""


@cli.group()
def ip():
    pass


@ip.group()
def floating():
    pass


@floating.command()
def bar():
    """Is a ip floating command"""


@ip.group()
def static():
    pass


@static.command()
def foo():
    """Is a ip static command"""
```

**setup.py**

```
from distutils.core import setup

setup(
    name='sampleclick',
    packages=['sampleclick'],
    entry_points={
        'console_scripts': [
            'sc = sampleclick.shell:cli'
        ]
    },
    install_requires=[
        'click>=6.6',
    ]
)
```

It is a bug in library or i doing something wrong?
I tryed search some help on [ST](http://stackoverflow.com/questions/37780146/click-module-autocomplete-does-not-work-at-a-depth-of-3-groups), but it seems no one is faced with similar situation.

**Comment by kirkhansen** [comment]
I'm also seeing this behavior with click version 6.6.

**Comment by johnarnold** [comment]
+1

**Comment by kirkhansen** [comment]
It looks like they've submitted a fix and tests for this a bit ago, and it's merged into master. Just waiting for the next release.
See #630

**Comment by balajeetm** [comment]
+1 I'm facing the same issue as well

**Comment by balajeetm** [comment]
Its been an year since it was raised and issue is still open. Is there any workarounds for this?

**Comment by kirkhansen** [comment]
@balajeetm  This works for me in version 6.7.

**Comment by jleveque** [comment]
I encountered this bug in version 6.6. I can attest that it is fixed for me in version 6.7 by commit https://github.com/pallets/click/commit/8af7ec14f3d936e02c2e4ee62d3573593868e07e.
